Scope success criteria in results

douglasjacobsen opened the pull request:
This merge adds a scoped name to the success criteria in the results output. The primary goal here is to help users understand where a criteria is coming from.

rfbgo commented on lib/ramble/ramble/success_criteria.py:
```diff
@@ -89,7 +91,8 @@ def passed(self):
         for scope in self._valid_scopes:
-            yield from self.criteria[scope]
+            for criteria in self.criteria[scope]:
+                yield criteria, scope
```
I don't actually mind this, but it seems like 90% of the calls here use the "old" interface and do not care about the scope we return. Do you think it's best to have a single condensed interface or two more explicit ones? 

I'm guessing the former, but wanted to check before merging

douglasjacobsen replied:
I don't really care much... but I also agree with you on the "most calls use the old interface". I figured adding the scope as an output here was more along the lines of DRY. I could also push the scope down into the criteria object instead, and then make it easier to extract.

rfbgo replied:
I'm not too worried either, just wanted to check in. If it's an intentional choice, then I'm happy

douglasjacobsen committed:
Scope success criteria in results

This commit adds a scoped name to the success criteria in the results
output. The primary goal here is to help users understand where a
criteria is coming from.
